server/controllers/profile.js

```diff
@@ -0,0 +1,118 @@
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+// @route POST /profile/create
+// @desc Create user profile with valid token
+// @access Private
+exports.createProfile = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+  const reqBody = req.body;
+
+  const profileExists = await Profile.exists({ userId: id });
+  if (profileExists) {
+    res.status(409);
+    throw new Error("The profile of this user has been created already");
+  }
+
+  const newProfile = await Profile.create({
+    userId: id,
+    firstName: reqBody.firstName,
+    lastName: reqBody.lastName,
+    gender: reqBody.gender,
+    birth: reqBody.birth,
+    phone: reqBody.phone,
+    address: reqBody.address,
+    description: reqBody.description,
+    availability: reqBody.availability,
+    photo: reqBody.photo,
+    isSitter: reqBody.isSitter,
+  });
+
+  if (!newProfile) {
+    res.status(500);
+    throw new Error("Server failed to create the new profile");
+  }
+
+  res.status(201).send({
+    message: "The new profile has been successfully created",
+    data: newProfile,
+  });
+});
+
+// @route GET /profile
+// @desc Get user profile with valid token
+// @access Private
+exports.getProfile = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+
+  if (id != req.body.userId) {
```

Comment:
It is not advisable to send information in the request body within a GET request :)